RELEASING.md

```diff
@@ -94,6 +94,7 @@ At this point you should evaluate the `git status` output as well as all the cha
     - `git commit -m 'merge 3.9 (changes unrelated to upgrade)'`
 3. Create **one** commit with the changes _customers need to apply in order to ugprade_, i.e. the image tag changes, adding/removing any new services, updating env vars, but no unrelated changes.
```

Comment:
These instructions might need a bit more adjustment now that we enforced pull requests on this repo for all branches. Not relevant to this PR, just noting it!

Reply:
@caugustus-sourcegraph 
Ahh, I see. Kevin and I ran into it during the 3.35.x patch release and we didn't know the change was intentional.

We ended up adding a new branch rule to bypass the branch protection rule on `x.x-customer-replica` branch https://github.com/sourcegraph/deploy-sourcegraph-docker/settings/branches

I'll go remove it then. Also we need update the docs to reflect the proper process as well.